svc/quiz/service.go

```diff
@@ -124,22 +128,38 @@ func (s *Service) GetQuizLink(ctx context.Context, uid uuid.UUID, username strin
 	s.mutex.Lock(key, time.Second*3)
 	defer s.mutex.Unlock(key)
 
+	receivedReward, err := s.challenges.GetChallengeReceivedRewardAmount(ctx, uid, challengeID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get received reward amount: %w", err)
+	}
+	if receivedReward > 0 {
+		return nil, fmt.Errorf("reward has been received for this challenge already: %w", err)
```

Comment:
in this case, the `err` variable is always `nil`
```suggestion
        return nil, errors.New("reward has been already received for this challenge")
```